### ISSUE TITLE
Intermittent 2+ second delays when typing

### ISSUE THREAD

**Issue by sabrons** [atom/atom#17257]
After returning to an Atom session that's been open for a few days, Atom feels terribly slow. About every 4-10 keystrokes, it will freeze for 1-3 seconds before continuing. It feels like a laggy 1980s teletype.

This isn't reproducible. Quitting and restarting made the problem go away.

I saved a trace, emailed to atom@github.com. It appears to be spawn calls causing the delay?

I'm glad to finally see an example of the catastrophic performance others have complained about. Until now, Atom has been snappy for me!

MacOS High Sierra, 10.13.4

```
~$ atom --version
Atom    : 1.25.1
Electron: 1.7.11
Chrome  : 58.0.3029.110
Node    : 7.9.0
~$ apm --version
apm  1.19.0
npm  3.10.10
node 6.9.5 x64
atom 1.25.1
python 2.7.10
git 2.16.1
```

apm list --bare

```
atom-dark-syntax@0.29.0
atom-dark-ui@0.53.1
atom-light-syntax@0.29.0
atom-light-ui@0.46.1
base16-tomorrow-dark-theme@1.5.0
base16-tomorrow-light-theme@1.5.0
one-dark-ui@1.10.10
one-light-ui@1.10.10
one-dark-syntax@1.8.2
one-light-syntax@1.8.2
solarized-dark-syntax@1.1.4
solarized-light-syntax@1.1.4
about@1.8.0
archive-view@0.64.2
autocomplete-atom-api@0.10.7
autocomplete-css@0.17.5
autocomplete-html@0.8.4
autocomplete-plus@2.40.2
autocomplete-snippets@1.12.0
autoflow@0.29.3
autosave@0.24.6
background-tips@0.27.1
bookmarks@0.45.1
bracket-matcher@0.89.1
command-palette@0.43.5
dalek@0.2.1
deprecation-cop@0.56.9
dev-live-reload@0.48.1
encoding-selector@0.23.8
exception-reporting@0.43.1
find-and-replace@0.215.5
fuzzy-finder@1.7.5
github@0.10.3
git-diff@1.3.9
go-to-line@0.33.0
grammar-selector@0.49.9
image-view@0.62.4
incompatible-packages@0.27.3
keybinding-resolver@0.38.1
line-ending-selector@0.7.5
link@0.31.4
markdown-preview@0.159.20
metrics@1.2.6
notifications@0.70.5
open-on-github@1.3.1
package-generator@1.3.0
settings-view@0.254.1
snippets@1.3.1
spell-check@0.72.7
status-bar@1.8.15
styleguide@0.49.10
symbols-view@0.118.2
tabs@0.109.1
timecop@0.36.2
tree-view@0.221.3
update-package-dependencies@0.13.1
welcome@0.36.6
whitespace@0.37.5
wrap-guide@0.40.3
language-c@0.59.2
language-clojure@0.22.7
language-coffee-script@0.49.3
language-csharp@1.0.1
language-css@0.42.10
language-gfm@0.90.3
language-git@0.19.1
language-go@0.45.2
language-html@0.49.0
language-hyperlink@0.16.3
language-java@0.28.0
language-javascript@0.128.3
language-json@0.19.1
language-less@0.34.2
language-make@0.22.3
language-mustache@0.14.5
language-objective-c@0.15.1
language-perl@0.38.1
language-php@0.43.1
language-property-list@0.9.1
language-python@0.49.2
language-ruby@0.71.4
language-ruby-on-rails@0.25.3
language-sass@0.61.4
language-shellscript@0.26.1
language-source@0.9.0
language-sql@0.25.10
language-text@0.7.3
language-todo@0.29.4
language-toml@0.18.2
language-typescript@0.3.2
language-xml@0.35.2
language-yaml@0.31.2

vim-mode-plus-replace-with-execution@0.4.0

advanced-open-file@0.16.8
atom-beautify@0.32.2
atom-ide-ui@0.9.3
atom-ternjs@0.18.3
autocomplete-paths@2.12.2
busy-signal@1.4.3
color-picker@2.2.5
cursor-history@0.13.1
docblockr@0.13.7
ex-mode@0.18.0
ex-mode-sort@1.0.0
file-icons@2.1.17
highlight-selected@0.13.1
ide-typescript@0.7.5
intentions@1.1.5
language-apache@1.8.0
language-babel@2.84.0
language-diff@0.7.0
language-gradle@0.0.5
language-haml@0.25.2
language-javascript-jsx@0.3.7
language-rust@0.4.12
language-slim@1.0.0
language-viml@1.1.6
linter@2.2.0
linter-clang@4.1.2
linter-coffeelint@1.3.1
linter-eslint@8.4.1
linter-haml@2.0.2
linter-js-standard@5.0.0
linter-js-yaml@1.2.8
linter-jsonlint@1.3.0
linter-rails-best-practices@0.2.2
linter-rubocop@2.2.3
linter-ruby@1.2.6
linter-rust@0.8.4
linter-slim@1.0.0
linter-tidy@2.3.1
linter-tslint@1.9.0
linter-ui-default@1.7.1
linter-write-good@0.9.0
narrow@0.64.0
oceanic-reef-syntax@0.9.0
open-this@0.5.0
package-sync@1.1.0
paner@1.0.0
prettier-atom@0.51.0
racer@0.20.0
rails-partials@0.9.2
rails-snippets@2.18.0
rails-transporter@1.7.1
rubocop-auto-correct@1.5.1
ruby-test-switcher@1.2.3
seti-syntax@1.1.3
seti-ui@1.10.0
tablr@1.8.3
tester@1.4.0
tester-jest@1.2.4
tester-mocha@1.0.0
todo-show@2.2.0
ubik-hackerman-syntax@1.10.0
vim-mode-plus@1.30.0
vim-mode-plus-jasmine-increase-focus@0.5.0
vim-mode-plus-keymaps-for-surround@0.2.1
vim-mode-plus-move-to-symbols@0.5.0
vim-mode-plus-replace-with-execution@0.4.0
white-cursor@0.7.0
```

**Comment by sabrons** [comment]
I *think* I started Atom from the Finder, so the current directory would be my gigantic home directory...   That may be useful information, maybe not.

Oops, I posted this from my work account. I normally do my Atom work as https://github.com/bronson

**Comment by Superdanby** [comment]
Maybe you ran into https://github.com/electron/electron/issues/11051.

**Comment by bronson** [comment]
@Superdanby, that's a good question.

I've included this profile's screenshot...  The canvas bug looks render-heavy, while this one looks scripting-heavy.

<img width="1157" alt="screen shot 2018-05-03 at 7 51 10 am" src="https://user-images.githubusercontent.com/1776/39584381-1fae8b4e-4ea7-11e8-9b39-a23e7a224717.png">

When I saved the trace, it looked like a `spawn` call was taking multiple seconds. Now that I've loaded it, I can't figure out what to click to show that.  So...  maybe that was a ghost.
